netmaster/master/netmaster.go

```diff
@@ -501,17 +501,26 @@ func CreateNetwork(network intent.ConfigNetwork, stateDriver core.StateDriver, t
 		// XXX: do configuration check, to make sure it is allowed
 	}
 
+	subnetIsAllocated := false
 	if nwCfg.SubnetIP == "" {
 		nwCfg.SubnetLen = gCfg.Auto.AllocSubnetLen
 		nwCfg.SubnetIP, err = gCfg.AllocSubnet(rm)
 		if err != nil {
 			return err
 		}
+		subnetIsAllocated = true
 	}
 
 	nwCfg.DefaultGw = network.DefaultGw
-	if nwCfg.DefaultGw == "" {
-		// TBD: allocate per global policy
+	// For auto derived subnets assign gateway ip be the last valid unicast ip the subnet
+	if nwCfg.DefaultGw == "" && subnetIsAllocated {
```

Comment:
> I think this more of a state modelling decision. I am fine tagging an endpoint with a flag or having a tenant level flag. I feel latter needs less error checking but am completely ok with either as long as we test :)

I understand tagging an endpoint as default, however I am not sure what you mean by 'tenant level flag'. How would, In your opinion, a tenant level flag be sufficient? Perhaps indicating that 'choose the first one as default' or 'choose the public-network leg as default'.
Can you shed some more light on what might be needed there?

In discussion with @shaleman it seems like we are covered for 'routed mode' forwarding because the subnet and gw are required elements there in the rest APIs. And it doesn't conflict with my change because the allocation of default gw only happens if subnet itself is dynamically allocated.

> I am completely ok if we are deciding to add them later. But usually it might be a good practice to back every functionality with tests otherwise we risk having breakages as more changes come in.

The ACI system testing is outside the scope of this change for sure. Because there are more elements needed in other repos that I am pushing the diffs in.